accounts-db/src/append_vec.rs

```diff
@@ -752,15 +760,20 @@ impl AppendVec {
                 } else {
                     // not enough was read from file to get `data`
                     assert!(data_len <= MAX_PERMITTED_DATA_LENGTH, "{data_len}");
-                    let mut data = vec![0u8; data_len as usize];
+                    let mut data: Box<[MaybeUninit<u8>]> = Box::new_uninit_slice(data_len as usize);
```

Comment:
Is it correct that only a certain "set" of threads call this function? If so, a follow-on PR could potentially:
- Increase the stack size for those threads (need `> MAX_PERMITTED_DATA_LENGTH`)
- Make this + the `Vec::with_capacity()` lower down in this same file utilize the stack as well

Certainly out of scope for this PR but mentioning it while we're thinking about this kind of stuff is present in a diff

Reply:
many followups are coming :)

A 10MB stack allocated array isn't a great idea especially since I think that appendvecs are loaded in parallel so would use 10*parallelism stack

Reply:
> many followups are coming :)

🔥 

> A 10MB stack allocated array isn't a great idea especially

After thinking about this some more, I think I'm in agreement with this sentiment. As Alessandro pointed out in DM's, having an API that is "This function can only be called by threads with X MB stack" is fairly brittle and a pretty big footgun with how we have things structured in Agave at the moment

Alessandro also mentioned concern with rayon-related recursion; I don't think we're susceptible to it in this specific case but it is another potential footgun that adds developer complexity.